dist/gcc-compatible/configure

```diff
@@ -100,6 +100,11 @@ detect_x64 () {
   [[ $target_arch == "x86_64" ]]
 }
 
+detect_x86 () {
+  [[ $target_arch == "i386" ]] || [[ $target_arch == "i686" ]] || \
```

Comment:
here too it feels like we're trying to match a lot of stuff; do you know if there are systems that may return `i86pc` or `ia32`? (just curious)